### PR TITLE
Accept a specific port in doctrine configuration

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | If a specific port is specified, it should be also added to doctrine configuration. |
| Type? | bug fix |
| Category? | IN |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-1055 |
| How to test? | Try an installation with a specific port, the entry `database_port` in file `app/config/parameters.yml` should be correctly updated. |

More, it fixes the installation in this specific use case :)

#### Important guidelines
- Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
- Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
- Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

### CODE REVIEW

**Comment by rGaillard** [comment]
@mickaelandrieu i don't have test, but i think the connection is also not working on sockets.

File: install-dev/models/install.php [hunk 1]
```diff
@@ -91,6 +91,12 @@ public function generateSettingsFile(
         $secret = Tools::passwdGen(56);
         $cookie_key = Tools::passwdGen(8);
         $cookie_iv = Tools::passwdGen(56);
+        
+        if (2 === count(preg_split('#:#', $database_host))) {
+            $splits = preg_split('#:#', $database_host);
```
Comment (rGaillard) [inline install-dev/models/install.php]: This will not work with an IPv6.

**Comment by mickaelandrieu** [comment]
@rGaillard maybe.. wont be part of this ticket but if an issue is spawned of course we will fix it :+1:

File: install-dev/models/install.php (hunk 1 above)
Comment (mickaelandrieu) [inline install-dev/models/install.php]: true, hmmm.. can you confirm PrestaShop is compatible with IPv6 ?

**Comment by Quetzacoalt91** [comment]
Follow-up: On its side, PrestaShop needs the port inside the `host` value.

**Comment by Quetzacoalt91** [comment]
Can you please rebase your PR in order to make the tests pass ?

**Comment by Quetzacoalt91** [comment]
Thanks dude

**Pull request merged**